src/goto-programs/interpreter.cpp

```diff
@@ -48,12 +91,128 @@ void interpretert::operator()()
   function=main_it;
     
   done=false;
-  
-  while(!done)
-  {
+  if(init) {
+    stack_depth=call_stack.size()+1;
     show_state();
-    command();
-    if(!done) step();
+    step();
+    while(!done && (stack_depth<=call_stack.size()) && (stack_depth>=0)) 
+    {
+      show_state();
+      step();
+    }
+    while(!done && (call_stack.size()==0))
+    {
+      show_state();
+      step();
+    }
+    clear_input_flags();
+    input_vars.clear();
+  }
+}
+
+std::set<irep_idt> interpretert::get_parameter_set(
+    const exprt &func_expr)
+{
+  typedef code_typet::parameterst parameterst;
+  typedef code_typet::parametert parametert;
+  const parameterst &params = to_code_type(func_expr.type()).parameters();
+
+  std::set<irep_idt> parameter_set;
+  for(const parametert & param : params)
+  {
+    parameter_set.insert(param.get_identifier());
+  }
+
+  return parameter_set;
+}
+
+/*******************************************************************\
+Function: interpretert::get_entry_function_call
+
+  Inputs: The Goto program being analysed
+
+ Outputs: The expression that represens the final call in the START
+          function
+
+ Purpose: To find the expression representing the first call of a
+          user provided function (e.g. the actual function being tested)
+
+\*******************************************************************/
+const code_function_callt &interpretert::get_entry_function_call(const goto_functionst &gf)
+{
+  typedef goto_functionst::function_mapt function_mapt;
```

Comment:
The reason why I did this was to keep under the 80 character limit for the line. Though for this line this doesn't seem to be a problem. Would you transform this line:

    const goto_functionst::function_mapt::const_iterator start_function = function_map.find(goto_start_id);

To a typedef of const_iterator, or just put a line break after the =?

Another case I was addressing the range based for loop in expr2c.cpp. I got:

    typedef struct_union_typet::componentt componentt;
    for(const componentt &component : struct_type.components())

Without the typedef, it reads like this:

    for(const struct_union_typet::componentt &component : struct_type.components())
                                                                                ^
                                                                   80 characters here

What approach should be taken here to keep under 80 characters?